src/cumin/pagination.clj

```diff
@@ -22,13 +22,14 @@
     (dec curr-page)))
 
 (defn- select-total [query]
-  (-> query
-      (dissoc ::pagination :post-queries :limit :offset :order)
-      (assoc :fields [])
-      (aggregate (count :id) :count)
-      (select)
-      (ffirst)
-      (last)))
+  (let [pk (get-in query [:ent :pk] :id)]
```

Comment:
Oops, I pushed too quick ;) Will update to use the `:pk` lookup and fall back to `:*`.